CMakeLists.txt

```diff
@@ -23,22 +23,48 @@ option(ALLOW_BENCHMARK              "ALLOW BENCHMARK"           OFF)
 option(RTTI_DISABLE                 "RTTI DISABLE"              ON)
 option(EXCEPTIONS_DISABLE           "EXCEPTIONS DISABLE"        ON)
 
+if (NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE Debug CACHE STRING "Build type (default Debug):" FORCE)
+endif()
+
+if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+
+    if (EXCEPTIONS_DISABLE)
+        message("Disable C++ exceptions.")
+        string(REGEX REPLACE "/EH[a-z]+" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc")
+        add_compile_definitions(__EXCEPTIONS_DISABLE__)
+    endif ()
+
+    if (RTTI_DISABLE)
+        message("Disable RTTI.")
+        string(REGEX REPLACE "/GR" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /GR-")
+    endif ()
+
+else (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
```

Comment:
if      (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
...
else (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")

What is this?

Reply:
This is looks odd.